app/src/main/java/org/inspire/breath/utils/RawToWavConverter.java

```diff
@@ -0,0 +1,106 @@
+package org.inspire.breath.utils;
+
+//        Credit: https://www.programcreek.com/java-api-examples/?code=timstableford%2FP-BrainAndroid%2FP-BrainAndroid-master%2Fapp%2Fsrc%2Fmain%2Fjava%2Fuk%2Fco%2Ftstableford%2Fp_brain%2FRecordWavMaster.java#
+
+//        The MIT License (MIT)
+//
+//        * * *
+//
+//        Permission is hereby granted, free of charge, to any person obtaining a copy of
+//        this software and associated documentation files (the “Software”), to deal in
+//        the Software without restriction, including without limitation the rights to
+//        use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+//        the Software, and to permit persons to whom the Software is furnished to do so,
+//        subject to the following conditions:
+//
+//        The above copyright notice and this permission notice shall be included in all
+//        copies or substantial portions of the Software.
+//
+//        THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//        IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+//        FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+//        COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+//        IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+//        CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.File;
+import java.io.DataInputStream;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public class RawToWavConverter {
+
+    public final int SAMPLE_RATE;
+
+    public RawToWavConverter(int sampleRate) {
+        this.SAMPLE_RATE = sampleRate;
+    }
+
+    private void writeString(final DataOutputStream output, final String value) throws IOException {
+        for (int i = 0; i < value.length(); i++) {
+            output.write(value.charAt(i));
+        }
+    }
+
+    private void writeInt(final DataOutputStream output, final int value) throws IOException {
+        output.write(value >> 0);
+        output.write(value >> 8);
+        output.write(value >> 16);
+        output.write(value >> 24);
+    }
+    private void writeShort(final DataOutputStream output, final short value) throws IOException {
+        output.write(value >> 0);
+        output.write(value >> 8);
+    }
+
+    /* Converting RAW format To WAV Format*/
+    public void rawToWave(final File rawFile, final File waveFile) throws IOException {
+
+        byte[] rawData = new byte[(int) rawFile.length()];
+        DataInputStream input = null;
+        try {
+            input = new DataInputStream(new FileInputStream(rawFile));
```

Comment:
Since we are just writing audio data to the FS it would probably make more sense to use [MediaRecorder](https://developer.android.com/guide/topics/media/mediarecorder) as this supports just putting it straight into a file. Also lossless. AudioRecorder is mainly for live data processing. Also media recorder exists from API level 1.

Reply:
I had been using MediaRecorder but I don't think lossless output is supported (https://developer.android.com/reference/android/media/MediaRecorder.OutputFormat.html), hence switching. Unless MediaRecorder is compatible with AudioRecord's output formats(?), I hadn't thought of that tbh.

Reply:
Ah cool, looks grand so :)